src/WinRT.Runtime/Projections/Nullable.cs

```diff
@@ -69,164 +174,326 @@ static unsafe BoxedValueIReferenceImpl()
+                IInspectableVftbl = Marshal.PtrToStructure<IInspectable.Vftbl>(vftblPtr.Vftbl);
+                get_Value_0 = GetValueDelegateForFunctionPointer(vftbl[6]);
+            }
+
+            internal static Delegate GetValueDelegateForFunctionPointer(IntPtr ptr)
+            {
+                if (typeof(T) == typeof(int) ||
+                    typeof(T) == typeof(byte) ||
+                    typeof(T) == typeof(bool) ||
+                    typeof(T) == typeof(sbyte) ||
+                    typeof(T) == typeof(short) ||
+                    typeof(T) == typeof(ushort) ||
+                    typeof(T) == typeof(char) ||
+                    typeof(T) == typeof(uint) ||
+                    typeof(T) == typeof(long) ||
+                    typeof(T) == typeof(ulong) ||
+                    typeof(T) == typeof(float) ||
+                    typeof(T) == typeof(double) ||
+                    typeof(T) == typeof(Guid) ||
+                    typeof(T) == typeof(global::System.TimeSpan) ||
+                    typeof(T) == typeof(global::Windows.Foundation.Point) ||
+                    typeof(T) == typeof(global::Windows.Foundation.Rect) ||
+                    typeof(T) == typeof(global::Windows.Foundation.Size) ||
+                    typeof(T) == typeof(global::System.Numerics.Matrix3x2) ||
+                    typeof(T) == typeof(global::System.Numerics.Matrix4x4) ||
+                    typeof(T) == typeof(global::System.Numerics.Plane) ||
+                    typeof(T) == typeof(global::System.Numerics.Quaternion) ||
+                    typeof(T) == typeof(global::System.Numerics.Vector2) ||
+                    typeof(T) == typeof(global::System.Numerics.Vector3) ||
+                    typeof(T) == typeof(global::System.Numerics.Vector4) ||
+                    (typeof(T).IsEnum && Enum.GetUnderlyingType(typeof(T)) == typeof(int)) ||
+                    (typeof(T).IsEnum && Enum.GetUnderlyingType(typeof(T)) == typeof(uint)))
+                {
+                    return Marshal.GetDelegateForFunctionPointer<Nullable_Delegates.GetValueDelegateAbi>(ptr);
+                }
+
+                if (typeof(T) == typeof(DateTimeOffset))
+                {
+                    return Marshal.GetDelegateForFunctionPointer<Nullable_Delegates.GetValueDelegateAbiDateTimeOffset>(ptr);
+                }
+
+#if NET
+                // Only when not on NAOT, use LINQ expressions to get the delegate type.
+                // This is not safe on AOT, so in that case we just can't get a delegate.
+                if (RuntimeFeature.IsDynamicCodeCompiled)
+#endif
+                {
+                    get_Value_0_Type ??= Projections.GetAbiDelegateType(typeof(void*), Marshaler<T>.AbiType.MakeByRefType(), typeof(int));
+
+                    return Marshal.GetDelegateForFunctionPointer(ptr, get_Value_0_Type);
+                }
+
+                throw new NotSupportedException("Failed to get the marshalling delegate for Nullable`1.");

```

Comment:
I think we will need to improve this scenario a bit here, but we can defer to another PR (possibly as part of mines) as it would break now anyways at runtime.